articles/ml_system_design/real_time_news_search_using_upstash_kafka_and_vector/upstash_ingest/upstash_ingest/cleaners.py

```diff
@@ -0,0 +1,52 @@
+from unstructured.cleaners.core import (
+    clean,
+    clean_non_ascii_chars,
+    replace_unicode_quotes,
+    remove_punctuation,
+    clean_ordered_bullets,
+)
+import re
+
+
+def standardize_dates(text, placeholder="[DATE]"):
+    date_pattern = (
+        r"\b(?:\d{1,2}[-/]\d{1,2}[-/]\d{2,4}|\d{2,4}[-/]\d{1,2}[-/]\d{1,2})\b"
+    )
+    return re.sub(date_pattern, placeholder, text)
+
+
+def standardize_times(text, placeholder="[TIME]"):
+    time_pattern = r"\b(?:\d{1,2}:\d{2}(?::\d{2})?\s?(?:AM|PM)?)\b"
+    return re.sub(time_pattern, placeholder, text)
+
+
+def remove_html_tags(text):
+    html_tag_pattern = r"<[^>]+>"
+    return re.sub(html_tag_pattern, "", text)
+
+
+def normalize_whitespace(text):
+    return re.sub(r"\s+", " ", text).strip()
+
+
+def clean_text(text: str) -> str:
+    """
```

Comment:
maybe just `clean` as `text` is in the parameter?

Reply:
Agree

Reply:
Changed to `clean_full` as clean will conflict with unstructured.core.clean. 
And using only unstructured.core.clean (meaning, removing the clean_full method) will showcase little ETL